dashboard/app/models/levels/level.rb

```diff
@@ -1013,8 +1013,23 @@ def skill_identifiers
     skills.map {|skill| {id: skill.id, key: skill.key}}
   end
 
-  def skill_keys
-    skills.pluck(:key)
+  def remove_skill_key(skill_key)
+    return unless Rails.application.config.levelbuilder_mode
+    skill_keys = JSON.parse(properties['skill_keys'])&.reject! {|sk| sk == skill_key} if properties['skill_keys']
+    # If the skill_keys array is empty, remove the property entirely.
+    if properties['skill_keys'].blank?
+      properties.delete('skill_keys')
+    end
+    properties['skill_keys'] = skill_keys.to_json if skill_keys
```

Comment:
please simplify this a bit by taking advantage of the assumptions that 
- you can assign directly to `self.skill_keys`
- assigning a value of `nil` will remove the key from the properties hash

the details of how properties work can be gleaned from https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/models/concerns/serialized_properties_test.rb